src/Sylius/Bundle/CoreBundle/Processor/AllCatalogPromotionsProcessor.php

```diff
@@ -13,34 +13,21 @@
 
 namespace Sylius\Bundle\CoreBundle\Processor;
 
-use Sylius\Bundle\PromotionBundle\Provider\EligibleCatalogPromotionsProviderInterface;
-use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Bundle\CoreBundle\Commander\UpdateVariantsCommanderInterface;
+use Sylius\Component\Core\Repository\ProductVariantRepositoryInterface;
 
 final class AllCatalogPromotionsProcessor implements AllCatalogPromotionsProcessorInterface
 {
-    private CatalogPromotionClearerInterface $catalogPromotionClearer;
-
-    private CatalogPromotionProcessorInterface $catalogPromotionProcessor;
-
-    private EligibleCatalogPromotionsProviderInterface $catalogPromotionsProvider;
-
     public function __construct(
-        CatalogPromotionClearerInterface $catalogPromotionClearer,
-        CatalogPromotionProcessorInterface $catalogPromotionProcessor,
-        EligibleCatalogPromotionsProviderInterface $catalogPromotionsProvider,
+        private ProductVariantRepositoryInterface $productVariantRepository,
+        private UpdateVariantsCommanderInterface $commander
     ) {
-        $this->catalogPromotionClearer = $catalogPromotionClearer;
-        $this->catalogPromotionProcessor = $catalogPromotionProcessor;
-        $this->catalogPromotionsProvider = $catalogPromotionsProvider;
     }
 
     public function process(): void
     {
-        $this->catalogPromotionClearer->clear();
-        $eligibleCatalogPromotions = $this->catalogPromotionsProvider->provide();
+        $variantsCodes = $this->productVariantRepository->getCodesOfAllVariants();
 
-        foreach ($eligibleCatalogPromotions as $catalogPromotion) {
-            $this->catalogPromotionProcessor->process($catalogPromotion);
-        }
+        $this->commander->updateVariants($variantsCodes);
```

Comment:
It's called `AllCatalogPromotionsProcessor` but what it does is getting codes of all **variants** and updating them. Seems like not catalog promotions related to me 😄 but maybe the problem is in naming of other services